cql3/statements/select_statement.cc

```diff
@@ -1396,6 +1567,18 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
     ::shared_ptr<cql3::statements::select_statement> stmt;
     auto prepared_attrs = _attrs->prepare(db, keyspace(), column_family());
     prepared_attrs->fill_prepare_context(ctx);
+
+    // Used to determine if an execution of this statement can be parallelized
+    // using `forward_service`.
+    auto can_be_forwarded = [&] {
+        return selection->is_aggregate()        // Aggregation only
+            && selection->is_count()            // Only count(*) selection is supported.
+            && !restrictions->need_filtering()  // No filtering
+            && group_by_cell_indices->empty()   // No GROUP BY
+            // All potential intermediate coordinators must support forwarding
+            && db.features().cluster_supports_parallelized_aggregation();
```

Comment:
A small problem is that a prepared statement won't be re-prepared once the cluster is upgraded, and will keep using the old implementation. I think it's tolerable.